device/uapi.go

```diff
@@ -420,7 +420,14 @@ func (device *Device) IpcHandle(socket net.Conn) {
 
 	switch op {
 	case "set=1\n":
-		status = device.IpcSetOperation(buffered.Reader)
+		err := device.IpcSetOperation(buffered.Reader)
+		if err != nil {
+			status, _ = err.(*IPCError)
```

Comment:
Isn't there a new Go 1.13 error API for this?

Reply:
There is. In 1.13 this could be:

```
var status *IPCError
if !errors.As(err, &status) {
  device.log.Error.Println("Invalid UAPI error:", err)
  status = &IPCError{1}
}
...
```